### ISSUE TITLE
Binary string escape sequences

### ISSUE THREAD

**Issue by wesboyt** [amzn/ion-js#165]
U0001D11E
text is accurate, bugged on binary.
턞 does not equal �

**Comment by pbcornell** [comment]
skiplist entry:  `good/equivs/utf8/stringU0001D11E.ion`

**Comment by zslayton** [comment]
I enabled that skiplist entry to see what we're up against. The equivs test that fails:
1. Reads the values in from a text Ion file (`good/equivs/utf8/stringU0001D11E.ion`)
2. Writes those values out into two byte arrays: one containing text Ion and one containing binary Ion.
3. Compares all of the values in the output within and across formats, expecting equivalence.

**binary/binary comparison**
```
AssertionError: Expected [0][0] e0 01 00 ea 83 ed 84 9e to be equivalent to [0][2] e0 01 00 ea 84 f0 9d 84 9e
```

**binary/text comparison**
```
AssertionError: Expected [0][0] e0 01 00 ea 83 ed 84 9e to be equivalent to [0][2] "ð"
```

**text/text comparison**
```
AssertionError: Expected [0][0] "í" to be equivalent to [0][2] "ð"
```
> text is accurate, bugged on binary.

Note that all of the comparisons fail, including `text/text`. I imagine that the issue lies in the initial text Ion reader reading from `good/equivs/utf8/stringU0001D11E.ion`, a phase of the process that all of the failing tests have in common.

**Comment by zslayton** [comment]
Fixed by #527.
